src/commands/infra/scaffold.ts

```diff
@@ -5,9 +5,86 @@ import yaml from "js-yaml";
 import path from "path";
 import { Config } from "../../config";
 import { logger } from "../../logger";
+import { IConfigYaml } from "../../types";
 import { validateRemoteSource } from "./generate";
 import * as infraCommon from "./infra_common";
 
+interface ICommandOptions {
+  [key: string]: string;
+}
+
+export const validateValues = (
+  config: IConfigYaml,
+  opts: ICommandOptions
+): boolean => {
+  if (
+    !config.azure_devops ||
+    !config.azure_devops.access_token ||
+    !config.azure_devops.infra_repository
+  ) {
+    logger.warn(
+      "The infrastructure repository containing the remote terraform template repo and access token was not specified. Checking passed arguments."
+    );
+
+    if (!opts.name || !opts.source || !opts.version || !opts.template) {
+      logger.error(
+        "You must specify each of the variables 'name', 'source', 'version', 'template' in order to scaffold out a deployment."
+      );
+      return false;
+    }
+    logger.info(
+      "All required options are configured via command line for scaffolding, expecting public remote repository for terraform templates or PAT embedded in source URL."
+    );
+  }
+  return true;
+};
+
+// Construct the source based on the the passed configurations of spk-config.yaml
+export const constructSource = (config: IConfigYaml) => {
+  const devops = config.azure_devops!;
+  const source = `https://spk:${devops.access_token}@${devops.infra_repository}`;
+  logger.info(
+    `Infrastructure repository detected from initialized spk-config.yaml.`
+  );
+  return source;
+};
+
+export const execute = async (
+  config: IConfigYaml,
+  opts: ICommandOptions,
+  exitFn: (status: number) => void
+) => {
+  if (!validateValues(config, opts)) {
+    exitFn(1);
+  } else {
+    opts.source = opts.source || constructSource(config);
+
+    try {
+      /* scaffoldDefinition will take in a definition object with a
+          null configuration. Hence, the first index is "" */
+      const scaffoldDefinition = ["", opts.source, opts.template, opts.version];
+      const sourceFolder = await infraCommon.repoCloneRegex(opts.source);
+      const sourcePath = path.join(infraCommon.spkTemplatesPath, sourceFolder);
+      await validateRemoteSource(scaffoldDefinition);
+      await copyTfTemplate(
+        path.join(sourcePath, opts.template),
+        opts.name,
+        false
+      );
+      await validateVariablesTf(
+        path.join(sourcePath, opts.template, "variables.tf")
+      );
+      await scaffold(opts.name, opts.source, opts.version, opts.template);
+      removeTemplateFiles(opts.name);
+      exitFn(0);
+    } catch (err) {
+      logger.error("Error occurred while generating scaffold");
+      logger.error(err);
+      exitFn(1);
+    }
+  }
+};
+
```

Comment:
Somewhere in this refactoring is breaking the functionality of `spk infra scaffold`. After testing the code, I am unable to generate a ` definition.yaml` file. I tried debugging, but am unable to find the root cause. Will need to dive into it a little deeper...

But you if you `spk infra scaffold` against the master branch, and one against your forked branch, you'll see that the in the refactored code, scaffolding fails to generate the definition and remove the templated terraform files.

SPK master branch:
<img width="710" alt="Screen Shot 2020-01-22 at 8 57 30 AM" src="https://user-images.githubusercontent.com/16811248/72915616-71ac6900-3cf5-11ea-80de-f33c828afa2c.png">

Your SPK master branch:
<img width="739" alt="Screen Shot 2020-01-22 at 8 58 33 AM" src="https://user-images.githubusercontent.com/16811248/72915639-7a9d3a80-3cf5-11ea-87ac-6c26eabab8e2.png">

Are you able to reproduce the same behavior on your side?

Reply:
chat with Yvonne and figure out the problem.